bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -152,16 +152,19 @@ func (r *KubeadmConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		if apierrors.IsNotFound(err) {
 			return ctrl.Result{}, nil
 		}
+		// Error reading the object  - requeue the request.
+		log.Error(err, "Failed to fetch KubeadmConfig")
```

Comment:
Just a quick first comment. We intentionally try to not log & return errors. The problem is that doing this leads to duplicate error messages in the logs (especially if this is done across multiple levels of a call stack)

(there are some exceptions, but only if we think there is some information that we should immediately surface (e.g. additional key value pairs from the logger)

Reply:
OK, I'd remove the log later.

> The problem is that doing this leads to duplicate error messages in the logs

I met the this problem many times and I was about to repeat same thing. Thank you for your good review.